src/Appwrite/Platform/Tasks/Restore.php

```diff
@@ -0,0 +1,291 @@
+    {
+        return 'restore';
+    }
+
+    public function __construct()
+    {
+        $this->checkEnvVariables();
+
+        $this
+            ->desc('Restore a DB')
+            ->param('id', '', new Text(20), 'Folder Identifier')
+            ->param('cloud', null, new WhiteList(['true', 'false'], true), 'Take file from cloud?')
+            ->param('database', null, new Text(10), 'example: db_fra1_01..')
+            ->param('datadir', null, new Text(100), 'mysql datadir path')
+            ->callback(fn ($id, $cloud, $project, $datadir) => $this->action($id, $cloud, $project, $datadir));
+    }
+
+    public function action(string $id, string $cloud, string $database, string $datadir): void
+    {
+        $this->database = $database;
+        $this->dsn = Backup::getDsn($database);
+
+        if (!$this->dsn instanceof DSN) {
+            Console::error('No dsn match');
+            Console::exit();
+        }
+
+        if (!file_exists($datadir)) {
+            Console::error('Datadir not found: ' . $datadir);
+            Console::exit();
+        }
+
+        if (file_exists($datadir . '/sys') || file_exists($datadir . '/appwrite')) {
+            Console::error('Datadir ' . $datadir . ' must be empty!');
+            Console::exit();
+        }
+
+        //todo: check if the container is stopped
+        //$this->checkContainerIsStopped();
+
+        $filename = $id . '.tar.gz';
+        Backup::log('--- Restore Start ' . $filename . ' --- ');
+        $start = microtime(true);
+        $cloud = $cloud === 'true';
+
+        if ($cloud) {
+            $local = new Local(Backup::BACKUPS . '/downloads/' . $id);
+
+            if (file_exists($local->getRoot())) {
+                $stdout = '';
+                $stderr = '';
+                $cmd = 'rm -rf ' . $local->getRoot();
+                Backup::log($cmd);
+                Console::execute($cmd, '', $stdout, $stderr);
+                if (!empty($stderr)) {
+                    Console::error($stderr);
+                    Console::exit();
+                }
+            }
+
+            $files = $local->getRoot() . '/files';
+
+            Backup::log('Creating Directory: ' . $files);
+            if (!mkdir($files, 0755, true)) {
+                Console::error('Error creating directory: ' . $files);
+                Console::exit();
+            }
+
+            $this->download($filename, $local);
+            $this->untar($local->getPath($filename), $files);
+        } else {
+            $local = new Local(Backup::BACKUPS . '/' . $database . '/full/' . $id);
+            $files = $local->getRoot() . '/files';
+        }
+
+        if (!file_exists($files)) {
+            Console::error('Directory not found: ' . $files);
+            Console::exit();
+        }
+
+        $this->decompress($files);
+        $this->prepare($files);
+        $this->restore($files, $cloud, $datadir);
+
+        Backup::log("Restore Finish in " . (microtime(true) - $start) . " seconds");
```

Comment:
Let's use single quotes for the strings here

Reply:
Done!